Tests/Unit/EmogrifierTest.php

```diff
@@ -1861,4 +1861,57 @@ public function dataUrisAreConserved()
             $this->subject->emogrify()
         );
     }
+
+    /**
+     * Data provider for css to html mapping.
+     *
+     * @return array[]
+     */
+    public function cssToHtmlMappingDataProvider()
+    {
+        return [
+            'Map background-color to bgcolor'
+                => ['', 'body {background-color: red;}', 'body', 'bgcolor="red"'],
+            'Do not map background URL'
+                => ['', 'body {background: url(bg.png) top;}', 'body', 'style'],
+            'Map background to bgcolor'
+                => ['', 'body {background: red top;}', 'body', 'bgcolor="red"'],
+            'Map width and height'
+                => ['', 'body {width: 100%; height: 200px;}', 'body', 'width="100%" height="200"'],
+            'Map text-align:center to align=center'
+                => ['<p>mid</p>', 'p {text-align: center;}', 'p', 'align="center"'],
+            'Do not map text-align: inherit'
+                => ['<p>plain</p>', 'p {text-align: interit;}', 'p', 'style'],
+            'Map margin: 0 auto to align=center'
+                => ['<img>', 'img {margin: 0 auto;}', 'img', 'align="center"'],
+            'Map float: right to align=right'
+                => ['<img>', 'img {float: right;}', 'img', 'align="right"'],
+            'Map border: none to border=0'
+                => ['<img>', 'img {border: none;}', 'img', 'border="0"'],
+            'Map border-spacing: 0 to cellspacing=0'
+                => ['<table><tr><td></td></tr></table>', 'table {border-spacing: 0;}', 'table', 'cellspacing="0"']
```

Comment:
For CSS properties that are limited to certain HTML elements or certain values, we also need one test (or data provider line) for each allowed element and value, and also one each for a case where neither applies, i.e., where the style is not applied.